public/app/plugins/datasource/elasticsearch/query_builder.ts

```diff
@@ -419,4 +421,64 @@ export class ElasticQueryBuilder {
       aggs: this.build(target, null, querystring).aggs,
     };
   }
+
+  addPPLAdhocFilters(queryString: any, adhocFilters: any) {
+    let i, adhocquery;
+    let filter = [];
+
+    for (i = 0; i < adhocFilters.length; i++) {
+      filter.push('`' + adhocFilters[i].key + '`');
+      if (dateMath.isValid(adhocFilters[i].value)) {
+        const validTime = dateTime(adhocFilters[i].value).format('YYYY-MM-DD HH:mm:ss.SSSSSS');
+        let timeFilter = "timestamp('$validTime')";
+        timeFilter = timeFilter.replace(/\$validTime/g, validTime);
+        filter.push(timeFilter);
+      } else if (typeof adhocFilters[i].value === 'string') {
+        filter.push("'" + adhocFilters[i].value + "'");
```

Comment:
Template strings might be cleaner for a lot of strings in this method. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals